### ISSUE TITLE
No "referenced but not assigned" in `values+=("$valeu")`

### ISSUE THREAD

**Issue by kbulygin** [koalaman/shellcheck#1791]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

set -eu

echo "$x"

opts=()
opts+=("$y")  # line 8

echo "$y"
```

#### Here's what shellcheck currently says:
```
Line 5	SC2154: x is referenced but not assigned.
```

#### Here's what I wanted or expected to see:
```
Line 8	SC2154: y is referenced but not assigned.
```
or at least
```
Line 10	SC2154: y is referenced but not assigned.
```
